dataset/src/main/scala/frameless/Job.scala

```diff
@@ -22,8 +22,17 @@ sealed abstract class Job[A](implicit sc: SparkContext) { self =>
       }
     }
   }
+
+  def map[B](fn: A => B): Job[B] = new Job[B]()(sc) {
+    def run(): B = fn(Job.this.run())
+  }
+
+  def flatMap[B](fn: A => Job[B]): Job[B] = new Job[B]()(sc) {
+    def run(): B = fn(Job.this.run()).run()
+  }
```

Comment:
The cats module doesn't depend on the dataset module, so we would have to make a `frameless-dataset-cats` module which doesn't really seem worth it. Or make `frameless-dataset` depend on cats (which it probably will eventually anyway)